android/library/src/main/java/com/liferay/mobile/screens/thingscreenlet/screens/ThingScreenlet.kt

```diff
@@ -165,32 +168,29 @@ open class ThingScreenlet @JvmOverloads constructor(
 		isSaveEnabled = true
 	}
 
+	private fun getApioAuthenticator(credentials: String? = null): ApioAuthenticator? {
+		val credentials = credentials ?: SessionContext.getCredentialsFromCurrentSession()
+
+		return credentials?.let {
+			BasicAuthenticator(credentials)
+		}
+	}
+
 	override fun onSaveInstanceState(): Parcelable {
+		var savedState = ThingScreenletSavedState(super.onSaveInstanceState())
+
 		thing?.let {
-			val bundle = Bundle()
-			bundle.putParcelable("superState", super.onSaveInstanceState())
-			bundle.putParcelable("thing", it)
-			return bundle
+			savedState.thing = it
 		}
 
-		return super.onSaveInstanceState()
+		return savedState
 	}
 
 	override fun onRestoreInstanceState(state: Parcelable?) {
-		if (state is Bundle) {
-			thing = state.getParcelable("thing")
-			super.onRestoreInstanceState(state.getParcelable("superState"))
-		} else {
-			super.onRestoreInstanceState(state)
-		}
+		var savedState: ThingScreenletSavedState = state as ThingScreenletSavedState
```

Comment:
Kotlin has type inference you just need:
`var savedState = state as ThingScreenletSavedState`